### DISCUSSION TITLE
Closure error

### DISCUSSION THREAD

**Discussion by eyelash** [vlang/v#11666] [Ideas]
**V version:** V 0.1.27 88ef0a7
**OS:** Linux

**What did you do?**
Since the changelog for V 0.1.27 mentions closures I did a very simple test:
```vlang
fn make_adder(x int) (fn(int)int) {
    return fn(y int) int {
        return x + y
    }
}
fn main() {
    adder := make_adder(3)
    println(adder(4))
}
```

**What did you expect to see?**
I expected the program to compile successfully and print `7`.
 
**What did you see instead?**
An error from the C compiler:
<pre>==================
/home/elias/.cache/v/sandbox.tmp.c: In function ‘anon_312_7_7’:
/home/elias/.cache/v/sandbox.tmp.c:1123:10: <font color="#CC0000">error:</font> ‘x’ undeclared (first use in this function)
 1123 |   return x + y;
      |          ^
/home/elias/.cache/v/sandbox.tmp.c:1123:10: note: each undeclared identifier is reported only once for each function it appears in
/home/elias/.cache/v/sandbox.tmp.c: In function ‘_STR’:
/home/elias/.cache/v/sandbox.tmp.c:1243:14: warning: pointer targets in passing argument 1 of ‘tos2’ differ in signedness [-Wpointer-sign]
 1243 |  return tos2(buf);
      |              ^~~
      |              |
      |              char *
/home/elias/.cache/v/sandbox.tmp.c:1002:21: note: expected ‘byteptr’ {aka ‘unsigned char *’} but argument is of type ‘char *’
...
==================
(Use `v -cg` to print the entire error message)

<font color="#CC0000"><b>builder error</b></font>: C error.</pre>

**Comment by eyelash** [comment]
BTW the syntax
```vlang
println(make_adder(3)(4))
```
results in the error
<pre><b>sandbox.v:35:23:</b> <font color="#CC0000"><b>error:</b></font> unexpected `(`, expecting `,` 
   33 | 
   34 | fn main() {
   35 | <font color="#CC0000">    println(make_adder(3)(4))</font>
      |                          <font color="#3465A4"><b>^</b></font>
   36 | }
</pre>

**Comment by funnywwh** [comment]
me too

**Comment by funnywwh** [comment]
```vlang
fn test(){
	t := 'text'
	f := fn (){
		println(t)
	}
	f()
	println(t)
}
fn main(){
	test()
}
```

```
v run .\test_cloure.v
==================
AppData\Local\Temp\v\test_cloure.tmp.c: In function 'anon_59__1':
AppData\Local\Temp\v\test_cloure.tmp.c:1266:11: error: 't' undeclared (first use in this function)
   println(t);
           ^
AppData\Local\Temp\v\test_cloure.tmp.c:1266:11: note: each undeclared identifier is reported only once for each function it appnC:\Users\funny\AppData\Local\Temp\v\test_cloure.tmp.c: In function 'strconv__v_sprintf':
AppData\Local\Temp\v\test_cloure.tmp.c:3667:7: warning: variable 'th_separator' set but not used [-Wunused-but-set-variable]
  bool th_separator = false;
       ^
AppData\Local\Temp\v\test_cloure.tmp.c: In function 'print':
AppData\Local\Temp\v\test_cloure.tmp.c:4623:4: warning: passing argument 4 of 'WriteConsoleW' from incompatible pointer type [e]    WriteConsole(output_handle, wide_str, wide_len, &bytes_written, 0);

==================
```

**Comment by danieldaeschle** [comment]
Since it doesn not error currently but is an unimplemented feature I changed it from "bug" to "feature request".
